core/src/main/java/hudson/FilePath.java

```diff
@@ -1065,6 +1066,8 @@ private boolean installIfNecessaryFrom(@NonNull URL archive, @NonNull TaskListen
             return true;
         } catch (IOException e) {
             throw new IOException("Failed to install " + archive + " to " + remote, e);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
```

Comment:
you should never be throwing this.

Reply:
Instead now throwing the following
`throw new URISyntaxException(e.getInput(), e.getReason());`